.github/workflows/driver_ci.yaml

```diff
@@ -0,0 +1,201 @@
+          $command_to_run `pwd`/build/reports
+        env:
+          command_to_run: ${{ matrix.command-to-run }}
+
+  CSI-deployment-k8s_yamls_validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: CSI-deployment- k8s yamls validation
+        run: |
+          ./scripts/run_yamlcheck.sh
+  CSI-Build-and-push-images:
+    runs-on: ubuntu-latest
+    needs:
+      - CSI-controller-static_code_analysis
+      - Unit_testing_coverage
+      - CSI-deployment-k8s_yamls_validation
+    strategy:
+      matrix:
+        image_type: ['node', 'controller']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1.3.0
+      - uses: FranzDiebold/github-env-vars-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: 'csiblock1'
+          password: '${{ secrets.CSIBLOCK_DOCKERHUB_PASSWORD }}'
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-${{ matrix.image_type }}
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Create github branch parameter
+        id: tested_github_branch
+        run: |
+          docker_image_branch_tag=`echo $CI_ACTION_REF_NAME| sed 's|/|.|g'`
+          echo "::set-output name=docker_image_branch_tag::${docker_image_branch_tag}"
+      - name: Build and push ${{ matrix.image_type }}
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          platforms: linux/amd64,linux/ppc64le,linux/s390x
+          push: true
+          tags: |
+            csiblock1/ibm-block-csi-${{ matrix.image_type }}:${{ steps.tested_github_branch.outputs.docker_image_branch_tag }}_${{ github.run_number }}
+            csiblock1/ibm-block-csi-${{ matrix.image_type }}:${{ steps.tested_github_branch.outputs.docker_image_branch_tag }}
+            csiblock1/ibm-block-csi-${{ matrix.image_type }}:latest
+          file: Dockerfile-csi-${{ matrix.image_type }}
+          cache-from: type=local,src=/tmp/.buildx-${{ matrix.image_type }}
+          cache-to: type=local,dest=/tmp/.buildx-new-${{ matrix.image_type }}
+        # Temp fix
+        # CSI-3164
+        # https://github.com/docker/build-push-action/issues/252
+        # https://github.com/moby/buildkit/issues/1896
+      - name: Move cache
+        run: |
+          [[ -G /tmp/.buildx-${{ matrix.image_type }} ]] && rm -rf /tmp/.buildx-${{ matrix.image_type }}
+          mv /tmp/.buildx-new-${{ matrix.image_type }} /tmp/.buildx-${{ matrix.image_type }}
+    outputs:
+      driver_images_tag: "${{ steps.tested_github_branch.outputs.docker_image_branch_tag }}_${{ github.run_number }}"
```

Comment:
could we unify the [tag creation logic](https://github.com/IBM/ibm-block-csi-driver/blob/a538b5ae9ee1822ed702671aa09fdbd9965ce7c3/scripts/ci/build_push_images.sh#L11)?

Reply:
done

Reply:
I don't see it here. link?

Reply:
https://github.com/IBM/ibm-block-csi-driver/blob/0827d76d4a112fdde05d1e4593f3a557bbea8012/.github/workflows/driver_ci.yaml#L137

Reply:
still looks like a duplication of [the link I wrote](https://github.com/IBM/ibm-block-csi-driver/pull/344#discussion_r667772008)

Reply:
I changed to the way that you asked for, am I missing something?

Reply:
I didn't mean that the code should be the same, I meant that it should be the same code.
i.e. in 1 common place instead of duplicated